src/core/common/logging.hpp

```diff
@@ -993,10 +993,10 @@ extern "C" {
  *
  */
 #if OPENTHREAD_CONFIG_LOG_PLATFORM == 1
-#define otLogCritPlat(aInstance, aFormat, ...) otLogCrit(aInstance, OT_LOG_REGION_PLATFORM, aFormat, ##__VA_ARGS__)
-#define otLogWarnPlat(aInstance, aFormat, ...) otLogWarn(aInstance, OT_LOG_REGION_PLATFORM, aFormat, ##__VA_ARGS__)
-#define otLogInfoPlat(aInstance, aFormat, ...) otLogInfo(aInstance, OT_LOG_REGION_PLATFORM, aFormat, ##__VA_ARGS__)
-#define otLogDebgPlat(aInstance, aFormat, ...) otLogDebg(aInstance, OT_LOG_REGION_PLATFORM, aFormat, ##__VA_ARGS__)
+#define otLogCritPlat(aInstance, aFormat, ...) otLogCrit(&aInstance, OT_LOG_REGION_PLATFORM, aFormat, ##__VA_ARGS__)
+#define otLogWarnPlat(aInstance, aFormat, ...) otLogWarn(&aInstance, OT_LOG_REGION_PLATFORM, aFormat, ##__VA_ARGS__)
+#define otLogInfoPlat(aInstance, aFormat, ...) otLogInfo(&aInstance, OT_LOG_REGION_PLATFORM, aFormat, ##__VA_ARGS__)
+#define otLogDebgPlat(aInstance, aFormat, ...) otLogDebg(&aInstance, OT_LOG_REGION_PLATFORM, aFormat, ##__VA_ARGS__)
```

Comment:
Note that the difference between `otLog<>Plat` and others log macros is intentional.
`otLog<>Plat()` expects a pointer to instance  but other log macros expect a reference.
```
/**
 * @def otLogCritPlat
 *
 * This method generates a log with level critical for the Platform region.
 *
 * @param[in]  aInstance A pointer to the OpenThread instance.
 * @param[in]  aFormat   A pointer to the format string.
 * @param[in]  ...       Arguments for the format specification.
 *
 */

/**
 * @def otLogCritCoap
 *
 * This method generates a log with level critical for the CoAP region.
 *
 * @param[in]  aInstance    A reference to the OpenThread instance.
 * @param[in]  aFormat      A pointer to the format string.
 * @param[in]  ...          Arguments for the format specification.
 *
 */

````
IIRC, the reason was that platform may be C based and has a pointer to instance.

Reply:
Very cool that the inline documentation is accurate. Thanks for pointing that out—I'll fix my internal branches to use this API correctly. I've pushed a second commit to this PR to revert this change.